procgov/ProcessModule.cs

```diff
@@ -271,7 +271,7 @@ void AssignProcessToExistingJobObject(int processId, Win32Job job, bool checkIfA
 
         fixed (char* penv = GetEnvironmentString(session.AdditionalEnvironmentVars))
         {
-            var args = string.Join(" ", procargs).ToCharArray();
+            var args = string.Join(" ", procargs, "\0").ToCharArray();
```

Comment:
Firstly, thank you for spotting this error and for the PR. Now, regarding this line, I believe you meant: ` (string.Join(" ", procargs) + "\0")` which would be OK. However, we still might have problems with arguments containing a space. Therefore, I would prefer to have the same logic as in the `StartProcessAndAssignToJobObject` function:

```
var args = (string.Join(" ", procargs.Select((string s) => s.Contains(' ') ? "\"" + s + "\"" : s)) + '\0').ToCharArray();
```

Could you please update the PR?

Reply:
Thank you @lowleveldesign for the quick review. You are correct, my original fix was buggy. I have updated this to match the argument handling performed in `StartProcessAndAssignToJobObject`, as you suggested.

Reply:
Great, thanks.